app/com/linkedin/drelephant/spark/fetchers/SparkFetcher.scala

```diff
@@ -62,31 +64,31 @@ class SparkFetcher(fetcherConfigurationData: FetcherConfigurationData)
   }
 
   override def fetchData(analyticJob: AnalyticJob): SparkApplicationData = {
+    doFetchData(analyticJob) match {
```

Comment:
Maybe this match block is not needed? Just call `doFetchData` method which either returns SparkApplicationData or throws an exception if an error occurs.